include/vga.h

```diff
@@ -151,6 +152,11 @@ enum Drawmode { PART, DRAWLINE, EGALINE };
 
 enum class VGA_RATE_MODE { DEFAULT, HOST, CUSTOM };
 
+enum PixelsPerChar : int8_t {
+	Eight = 8,
+	Nine  = 9,
+};
```

Comment:
If we're using these enums to decide whether to use the 8x14 EGA font or the 9x16 VGA font in the DOSBox window by default, then this enum is a bit wrong (we're inferring which font and thus which screen mode to use from a "pixels per character" value; that's not great).

I'd convert this into this instead:

```cpp
enum Font : int8_t {
    EGA8x14 = 8,
    VGA9x16  = 9,
};
``` 

Personally, I wouldn't infer the screen mode to use from the font either (8x16 font -> 640x350, 9x16 -> 720x400), and would make the screen mode to use explicit in that struct too.

Reply:
Previously these were magic `8`s and `9`s.. and I had to use more of them, so wanted to de-magic.

They're used as quantities, for example, the clock frequency is divided by the number of pixels per character:

```c++
// Adjust the VGA clock frequency based on the Clocking Mode Register's
// 9/8 Dot Mode. See Timing Model: https://wiki.osdev.org/VGA_Hardware

		clock /= vga.seq.clocking_mode.is_eight_dot_mode
		               ? PixelsPerChar::Eight
		               : PixelsPerChar::Nine;
```

And used as quantities in other places too - 

```c++
const auto cursor_draw_offset = check_cast<uint16_t>(
        attr_addr * vga.draw.pixels_per_character * bytes_per_pixel);
```

where `pixels_per_character` is a `PixelsPerChar`.

But yeah - it's clunky - all open to renames!

Reply:
(I should also clarify that resolution isn't being inferred or assumed by this value.. it's only describing the font pixel-size properties)

Reply:
Yeah okay, seems low-level enough, so maybe leave it. Especially if there's a "9-pixels-per-char" VGA register bit somewhere too...

Reply:
>  Especially if there's a "9-pixels-per-char" VGA register bit somewhere too...

Yup, there is:

![2023-04-18_20-08](https://user-images.githubusercontent.com/1557255/232956851-6f202a18-8fe3-4a43-88e5-48b3a4b4cb86.png)

This register corresponds to `clocking_mode` 8-bit member and the 9/8 dot mode bit corresponds to the `is_eight_dot_mode` variable:

`vga.seq.clocking_mode.is_eight_dot_mode`

Which the `bitview<0, 1>` directly extracts from the register.